ils_middleware/tasks/amazon/alma_work_s3.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from urllib.parse import urlparse
+import os
+from os import path
+from airflow.models import Variable
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from pymarc import MARCReader
+from rdflib import Graph
+from lxml import etree as ET
+
+logger = logging.getLogger(__name__)
+
+
+def get_from_alma_s3(**kwargs):
+    s3_hook = S3Hook(aws_conn_id="aws_lambda_connection")
+    task_instance = kwargs.get("task_instance")
+    resources = task_instance.xcom_pull(key="resources", task_ids="sqs-message-parse")
+
+    for instance_uri in resources:
+        instance_path = urlparse(instance_uri).path
+        instance_id = path.split(instance_path)[-1]
+
+        temp_file = s3_hook.download_file(
+            key=f"marc/airflow/{instance_id}/record.mar",
+            bucket_name=Variable.get("marc_s3_bucket"),
+        )
+        task_instance.xcom_push(key=instance_uri, value=temp_file)
+
+
+def send_work_to_alma_s3(**kwargs):
+    s3_hook = S3Hook(aws_conn_id="aws_lambda_connection")
+    task_instance = kwargs.get("task_instance")
+    resources = task_instance.xcom_pull(key="resources", task_ids="sqs-message-parse")
+
+    for instance_uri in resources:
+        instance_path = urlparse(instance_uri).path
+        instance_id = path.split(instance_path)[-1]
+
+        temp_file = task_instance.xcom_pull(
+            key=instance_uri, task_ids="process_alma.download_marc"
+        )
+        marc_record_from_temp_file(instance_id, temp_file)
+        with open(temp_file, "rb") as marc_file:
+            reader = MARCReader(marc_file)
+            for record in reader:
+                work_field = record.get_fields("758")
+                work_uri = work_field[0].get_subfields("0")[0]
+                logger.info(f"Work URI: {work_uri}")
+            g = Graph()
+            g.parse(work_uri)
+            g.bind("xmlns", "http://www.w3.org/2000/xmlns/")
```

Comment:
See suggestions above for using a `for` loop on a common `alma_namespaces` variable.